docs/DeveloperGuide.md

```diff
@@ -1,38 +1,306 @@
+
+These operations are detailed in the ```Parser``` class, where the ```parseCommand(String userInput, UserDetails userDetails)``` method is updated to accept the ```UserDetails``` object and return a ```GenerateSmartItineraryCommand``` when the `"smart"` command is entered by the user.
+
+Given below is an example usage scenario and how the Smart Itinerary Generation mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. Their user details, including preferred location and cuisine preferences, are collected via the ```UserDetailsCommand``` and stored in a ```UserDetails``` object.
+
+Step 2. The user enters the `"smart"` command to generate a personalized itinerary. The `"smart"` command is parsed by the ```parseCommand``` method in the ```Parser``` class, which creates a ```GenerateSmartItineraryCommand``` instance, passing the ```UserDetails``` object.
+
+Step 3. The execute method of ```GenerateSmartItineraryCommand``` is invoked. It retrieves the user's preferred location and cuisine from the ```UserDetails``` object. The method then filters the ```FoodList``` to find food options matching the user's preferences, ensures the selected food options are distinct, and randomly selects two activities from the ```ActivityList```. The generated itinerary, consisting of the selected food and activity options, is then output to the user.
+
+![Smart Itinerary Generation Sequence Diagram](images/GenerateSmartItineraryCommandSequenceDiagram.png)
+
+### [Implemented] User Details Collection
+
+#### Implementation
+The UserDetailsCommand is responsible for collecting personal information from the user, which includes name, age, gender, location, favorite cuisine, and relationship status. This command may also conditionally request an anniversary date based on the user's relationship status. The information is then utilized to enhance the application's service and recommendations.
+
+This command extends the Command class and implements the following operations:
+
+`UserDetailsCommand#execute()` — Interacts with the user via the UI to gather personal information and saves it using the `Storage` class.
+Personal details such as name, age, and gender are stored in a `UserDetails` object, which is then serialized to a file by `Storage`.
+These operations are exposed in the system as follows:
+
+The `UserDetailsCommand` is instantiated and called when the user chooses to enter or update their personal details.
+The execute method of this command interacts with other components like `Ui` for input/output and `Storage` for persisting user data.
+Below is the sequence of actions performed by the `UserDetailsCommand` to collect user information.
+
+![User Details Collection Sequence Diagram](images/UserDetailsCommandSequenceDiagram.png)
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/75208150/1e83ecad-aedd-4d0a-95b3-00925a807c2b)
Is this the appropriate way of showing the different alternate paths taken? 
Rather a format like the picture below is better in showing the alternate paths taken.
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/75208150/86d59768-1381-4477-b4b8-d9933b01b4f6)

Reply:
Perhaps c:UserDetailsCommand could be moved to the left, while u:UI be in the middle, such that the diagram can flow from left to right